orc8r/cloud/go/service/middleware/unary/interceptors.go

```diff
@@ -76,29 +76,17 @@ func errlogInterceptor(ctx context.Context, req interface{}, info *grpc.UnarySer
 	return res, err
 }
 
-// setIdentityInterceptor sets the caller's identity from certificate metadata
-// in the caller's context.
-// TODO(hcgatewood): remove this wrapper and just use SetIdentityFromContext directly
-func setIdentityInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
-	newCtx, _, res, err := SetIdentityFromContext(ctx, req, info)
-	if err != nil {
-		return res, err
-	}
-	if newCtx != nil {
-		ctx = newCtx
-	}
-	return handler(ctx, req)
-}
+type interceptorTemplate func(context.Context, interface{}, *grpc.UnaryServerInfo) (context.Context, interface{}, interface{}, error)
 
-// BlockUnregisteredGateways blocks requests from unregistered gateways.
-// TODO(hcgatewood): remove this wrapper and just use BlockUnregisteredGateways directly
-func blockUnregisteredGatewaysInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
-	newCtx, _, res, err := BlockUnregisteredGateways(ctx, req, info)
-	if err != nil {
-		return res, err
-	}
-	if newCtx != nil {
-		ctx = newCtx
+func makeInterceptorFromTemplate(fn interceptorTemplate) (usi grpc.UnaryServerInterceptor) {
```

Comment:
Nit: You don't use the returned variable, I think this could be simplified.
```suggestion
func makeInterceptorFromTemplate(fn interceptorTemplate) grpc.UnaryServerInterceptor {
```

Reply:
Thanks!